client/src/SignUpShopper.js

```diff
@@ -0,0 +1,98 @@
+import { useEffect, useState } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import Api from './Api';
+
+//to add new component, make new file
+//to create (crud), use forms
+function SignUpShopper() {
+
+    //:id from Sections.js 
+    const {id} = useParams();
+    const history = useHistory();
+    
+    //initial value of section (in this case, obj with name, slug, and position attributes)
+    const [shopper, setShopper] = useState({
+        UserId: 0,
+        zipcode: 0
+    });
+
+    const [user, setUser] = useState({
+      photo: '',
+      birthday: 0,
+      gender: ''
+    });
+
+    //side effects, don't directly interact with output, don't refresh when it changes
+    useEffect(function(){
+        if(id){
+            Api.shoppers.get(id).then((response) => setShopper(response.data));
+            Api.users.get(id).then((response) => setUser(response.data));
+        }
+    }, []);
+
+    function onChange(event) {
+        //new object with current objects in section array
+        const newShopper = { ...shopper };
+        const newUser = {...user };
+        
+        //look for which name, and change the obj with that name with inputted 'value'
+        //modify model (section)
+        newShopper[event.target.name] = event.target.value;
+        newUser[event.target.name] = event.target.value;
+        setShopper(newShopper);
+        setUser(newUser);
+    }
+    //async function execute (multitasking) js continue running when server connects
+    async function onSubmit(event) {
+        event.preventDefault();
+
+        //might cause error, wrap in try
+        try {
+
+            //update
```

Comment:
We know we won't be "creating" a user here, that will have been done in the register step before this page. I think we can also make it such that the one-to-one Shopper record can be created for the User automatically as needed. So we can simplify this code to just updates. I'll help with this...